.github/workflows/ci.yml

```diff
@@ -839,6 +839,8 @@ jobs:
         uses: ./.github/actions/run-maven
         with:
           parameters: -f build-tools/pom.xml install
+      - name: Build Single App If Changed
+        uses: ./.github/actions/build-singleapp-if-changed
```

Comment:
We have to keep an eye that this additional step doesn't prolong the build time too much. In https://github.com/camunda/camunda/actions/runs/13905902394/job/38909053399?pr=29633 I see it adds nearly 4 minutes of runtime, given a 10min overall runtime goal and a 15min hard limit.